src/notifications/common.ts

```diff
@@ -235,3 +236,33 @@ export const getUnreadNotificationsCount = async (
       readAt: IsNull(),
     },
   });
+
+enum UserNotificationUniqueKey {
+  PostAdded = 'post_added',
+}
+
+const notificationTypeToUniqueKey: Partial<
+  Record<NotificationType, UserNotificationUniqueKey>
+> = {
+  [NotificationType.SquadPostAdded]: UserNotificationUniqueKey.PostAdded,
+  [NotificationType.SourcePostAdded]: UserNotificationUniqueKey.PostAdded,
+  [NotificationType.UserPostAdded]: UserNotificationUniqueKey.PostAdded,
+};
+
+export const generateUserNotificationUniqueKey = ({
+  type,
+  referenceId,
+  referenceType,
+}: {
+  type: NotificationType;
+  referenceId?: string;
+  referenceType?: NotificationReferenceType;
```

Comment:
I don't think we need `notification_v2.uniqueKey` here because `notificationId` on `user_notification` already dedups over it.